CMakeLists.txt

```diff
@@ -228,6 +228,11 @@ endif()
 
 ## Harfbuzz ##
 ##############
+
+if (APPLE)
+    option(HB_HAVE_CORETEXT "Enable CoreText shaper backend on macOS" ON)
```

Comment:
We should have that option at the top with other options, also `APPLE` is only defined is used from tangram-es, we should rather control that option being set in tangram-es based on the platform used e.g.:

https://github.com/tangrams/tangram-es/blob/master/core/deps/CMakeLists.txt#L54
```
if(APPLE)
  set(HB_HAVE_CORETEXT ON CACHE BOOL "Enable core text support")
endif()
```

Also we should document `HB_HAVE_CORETEXT` from the readme and rename it to `HARFBUZZ_HAVE_CORETEXT`.

Reply:
Moved the `option` above with other `option`s. Discussed the usage of `APPLE` on slack and `APPLE` is a cmake keyword and works aptly here.